app/uk/gov/hmrc/helptosavefrontend/controllers/UpdateEmailAddressController.scala

```diff
@@ -44,9 +49,29 @@ class UpdateEmailAddressController @Inject()(val sessionCacheConnector: SessionC
           } {
             _.eligibilityCheckResult.fold(
               Ok(views.html.core.not_eligible())
-            )(_ ⇒ Ok(views.html.register.update_email_address()))
+            )( userInfo ⇒ {
+              Ok(views.html.register.update_email_address(userInfo.contactDetails.email, Some(UpdateEmailForm.verifyEmailForm)))
+            })
           }
         }
   }(redirectOnLoginURL = FrontendAppConfig.checkEligibilityUrl)
 
-}
+  def onSubmit(): Action[AnyContent] = authorisedForHtsWithInfo {
+    implicit request =>
+      implicit htsContext ⇒
+        UpdateEmailForm.verifyEmailForm.bindFromRequest().fold(
+          formWithErrors => {
+            Future.successful(BadRequest(views.html.register.update_email_address("errors", Some(formWithErrors))))
+          },
+          (details: UpdateEmail) => {
+           emailVerificationConnector.verifyEmail(htsContext.nino.getOrElse(""), details.email).map {
+             case Right(x) ⇒ Ok(views.html.register.check_your_email())
+             case Left(AlreadyVerified()) ⇒ Ok(views.html.register.email_verify_error("hts.email-verification.email-verify-error.already-verified.content"))
```

Comment:
I think we should try to minimise the amount of code in views because the tooling is so primitive - certainly passing in message keys is a standard pattern in other projects.